src/core/thread/key_manager.cpp

```diff
@@ -199,13 +205,85 @@ void KeyManager::Stop(void)
     mKeyRotationTimer.Stop();
 }
 
+#if OPENTHREAD_CONFIG_PSA_CRYPTO_ENABLE
+
+#if OPENTHREAD_MTD || OPENTHREAD_FTD
+void KeyManager::SetPskc(const Pskc &aPskc)
+{
+    Error error = kErrorNone;
+
+    mPskcRef = kPkscPsaItsOffset;
```

Comment:
Is it correct? I think `mPskcRef` is initialized by `otPlatPsaImportKey` and the very same value should be passed to `CheckAndDestroyStoredKey`. This line causes that the original key ID is lost and can't be released when another pskc is set. The same issue appears in `StoreMasterKey` below.

Reply:
@Damian-Nordic PSKc and MasterKey are persistent keys in the system, and as such PSA will not assign key IDs to these keys. Instead the application provide the KeyID to be used. That is why it doesn't matter even if `mPskcRef` is overwritten, the value never changes. 

In the next update `mPskcRef` is being removed completely and instead the references are being stored in the `Pskc` class. I am still testing and will update the PR soon.

Reply:
According to `https://armmbed.github.io/mbed-crypto/html/api/keys/management.html#c.psa_import_key` `key` is an output parameter of `psa_import_key`, so I'm not sure if I understand your plan. Is `otPlatPsaImportKey` supposed to just call `psa_import_key` or do something else?

Reply:
Ok, I see. But why using fixed key IDs is better than letting `psa_import_key` generate ones? Fixed key IDs imply that we need to manually synchronize different libraries which use the PSA API.

Reply:
Well, `psa_import_key` is not designed to generate id for persistent keys. It expects the app to provide the keyID through key attributes. It is not a design decision but PSA constraint. 

I assume this is because of keyId management in app. If a random keyID is generated for persistent keys by PSA, app needs to store this keyID for further use and needs to make sure the ID persists between power cycles. If the app allocates a keyId it can be as simple as using macros or constants.

Reply:
Right, it looks like ranges for user-provided `<PSA_KEY_ID_USER_MIN, PSA_KEY_ID_USER_MAX>` and auto-generated `<PSA_KEY_ID_VENDOR_MIN, PSA_KEY_ID_VENDOR_MAX>` key IDs are disjoint, so using fixed keyIDs shouldn't be an issue and that way they don't need to be persistent. Makes sense to me now. Thanks for explanation.